src/utils.js

```diff
@@ -383,3 +383,44 @@ export function reassignKeys(track) {
         track[i].key = i;
     }
 }
+
+/**
+ * This function handles the drag stop event for track items.
+ *
+ * It takes a track array, the dragged grid item, and the sequence
+ * duration.
+ *
+ * It returns the new state of the track, which the caller can save
+ * with setState. If there are no necessary changes to the track,
+ * this function returns null.
+ */
+export function trackItemDragHandler(origTrack, item, sequenceDuration) {
+    // react-grid-layout sets a 'moved' flag on the dragged item.
+    if (item.moved === false) {
+        return null;
+    }
+
+    const origElem = _.find(origTrack, ['key', parseInt(item['i'], 10)]);
+    let newTrack = origTrack.slice();
+    const elem = _.find(newTrack, ['key', parseInt(item['i'], 10)]);
+    const percent = item.x / 1000;
+    const startTime = percent * sequenceDuration;
+
+    // Check that the value actually needs to be updated.
+    if (startTime === origElem.start_time ||
+        // Allow just over a hundredth of a second margin of
+        // error. This can occur when clicking the track elements.
+        Math.abs(startTime - origElem.start_time) <= 0.015
```

Comment:
I just realized this 0.015 value is dependent on the sequenceDuration - this should be larger for long videos and smaller for short ones.